app/assets/javascripts/proxy_status.js

```diff
@@ -0,0 +1,100 @@
+function setItemStatus(item, response) {
+  if(response.success) {
+    item.attr('title', __('Active'));
+    item.addClass('text-success');
+    item.html(icon('ok-sign'));
+  } else {
+    item.attr('title', response.message);
+    item.addClass('text-danger');
+    item.html(icon('exclamation-sign'));
+  }
+  item.tooltip({html: true});
+}
+
+function setProxyVersion(item, response) {
+  if(response.success) {
+    item.addClass('text-success');
+    item.text(response.message.version);
+  } else {
+    item.addClass('text-danger');
+    item.text(__('Error'));
+  }
+  item.tooltip({html: true});
+}
+
+function setPluginVersion(item, response) {
+  var pluginName = item.data('plugin');
+  var pluginVersion = response.message.modules ? response.message.modules[pluginName] : __('Unknown');
+  if(response.success) {
+    item.text(pluginVersion);
+  } else {
+    item.addClass('text-danger');
+    item.text(__('Error'));
+  }
+  item.tooltip({html: true});
+}
+
+$(function() {
+  $('.proxy-show').each(function(index, item) {
+    var proxy = new ProxyStatus($(item));
+    proxy.getVersions();
+  });
+
+  $('.proxy-tftp').each(function(index, item) {
+    var item = $(item);
+    var url = item.data('url');
+    $.ajax({
+      type: 'get',
+      url: url,
+      success: function (response) {
+        if(response.success) {
+          item.addClass('text-success');
+          item.text(response.message);
+        } else {
+          item.addClass('text-danger');
+          item.text(__('Error'));
+        }
+      },
+      error: function (response) {
+        item.addClass('text-danger');
+        item.text(__('Error'));
+      }
+    });
+  });
+});
+
+function ProxyStatus(item) {
+  this.url = item.data('url');
+  this.item = item;
+  var self = this;
+
+  this.getVersions = function() {
+    $.ajax({
+      type: 'get',
+      url: this.url,
+      success: function (response) {
+        populateData(response, self.item);
+      }.bind(this),
+      error: function (response) {
```

Comment:
Since the handling is the same for success and error, you can simply use `complete`.